lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -354,23 +366,57 @@ def add_path(self, path: str, *, recursive: bool = False) -> None:
         Returns:
             True if path was added else False.
         """
-        path = Path(path)  # type: Path
+        path = Path(path) if isinstance(path, str) else path  # type: Path
         if path.is_dir():
-            self.alert_groups.extend(self._from_dir(path, recursive))
+            self.groups.extend(self._from_dir(path, recursive))
         elif path.is_file():
-            self.alert_groups.extend(self._from_file(path.parent, path))
+            self.groups.extend(self._from_file(path.parent, path))
         else:
-            logger.debug("Alert rules path does not exist: %s", path)
+            logger.debug("%s rules path does not exist: %s", self.rule_type.capitalize(), path)
 
     def as_dict(self) -> dict:
-        """Return standard alert rules file in dict representation.
+        """Return standard rules file in dict representation.
 
         Returns:
             a dictionary containing a single list of alert rule groups.
             The list of alert rule groups is provided as value of the
             "groups" dictionary key.
         """
-        return {"groups": self.alert_groups} if self.alert_groups else {}
+        return {"groups": self.groups} if self.groups else {}
+
+
+class AlertRules(Rules):
+    """Utility class for amalgamating prometheus alert files and injecting juju topology.
+
+    The official Prometheus format is a YAML file conforming to the Prometheus documentation
+    (https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/).
+    The custom single rule format is a subsection of the official YAML, having a single alert
+    rule, effectively "one alert per file".
+    """
+
+    _rule_type = "alert"  # type: _RuleType
+
+    @property
+    def rule_type(self) -> _RuleType:
```

Comment:
Probably wouldn't need a property if we use ClassVar?

Reply:
Well, it would, because there's no similar ClassVar on the parent, so `self.rule_type` won't resolve on `AlertRules(ABC)` without at least an `@abstractmethod`, hence the property. Parents can't use the attributes of their children unless they're defined, and the ABC parent has no reason to have a `_rule_type`